### DISCUSSION TITLE
Decouple what is necessary to run aframe+NAF(w/ dialog Adapter) into another project

### DISCUSSION THREAD

**Discussion by massimo-bortolamei-hegias** [mozilla/hubs#3130] [Hubs Cloud Questions] [answered]
Hello, i'm customising the hubs client to leverage its aframe+networked-aframe(with dialog adapter). I noticed there is a lot of auth/communication with the Phoenix/Reticulum servers in hubs.js, on which the naf-dialog-adapter is reliant. Is there a way to isolate just what is necessary to retrieve jwt tokens, connect to the hubs servers (deployed on my aws) and have a totally custom aframe scene in order to embed it in my own project?
I can provide more details on what want to do, if needed
Any suggestion would be appreciated. 
Thanks

**Comment by robinkwilson** [comment, marked as answer]
Ah that sounds really cool what you're working on! We don't have that built into the system so it's going to be an undertaking to do. 

Unfortunately we don't really have the capacity to scope/closely help with that kind of project but we'd love to see what you come up with! If you have any specific specific questions about the system, feel free to make another thread. :)

**Reply by massimo-bortolamei-hegias** [reply]
Hi @robin-k-wilson thanks for answering!
Yes its very complicated, I agree !
I was able to understand the general flow , and I believe I'm very close to a first POC, but I'm blocked at connecting to the socket!
While the clean hubs-client just calls socket.connect() with no additional parameters and it receives a 101 upgrade response, i get a 403 forbidden message. If I could get help in understanding how to enable this socket connection, it would be great.

My feeling is that something important happens at npm run login time, which enables webpack to communicate with the socket and get authorized. I need to skip all that, and just understand how to manage to connect to it properly, and unfortunately there is no doc around.

If you prefer to talk about this in another thread, just tell me and I'll create it.
Thanks for your time,
Cheers
